File: WebApplications/Laravel/application_trial/app/Http/Resources/Opportunity.php
<?php

namespace App\Http\Resources;

use App\Http\Resources\Lookups\Category;
use App\Http\Resources\Lookups\Country;
use App\Models\Models\OpportunityDetail;
use Illuminate\Http\Request;
use Illuminate\Http\Resources\Json\JsonResource;

class Opportunity extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'title' => $this->title,
            'category' => $this->category,
            'country' => new Country($this->country),
            'deadline' => $this->deadline->toFormattedDateString(),
            'createdBy' => $this->user,
            'organizer' => $this->organizer,
            'opportunityDetails' => $this->detail,
            'createdAt' => $this->created_at->format("d.m.y H:i"),
            'updatedAt' => $this->created_at->toDateTimeString(),
        ];
    }
}

File: WebApplications/Laravel/forsat/lib/network/classes/opportunity/opportunity.dart
import 'package:forsat/network/classes/common/category.dart';
import 'package:forsat/network/classes/common/country.dart';
import 'package:forsat/network/classes/opportunity/opportunity_detail.dart';

class Opportunity {
  final int id;
  final String title;
  final Category category;
  final Country country;
  final String deadline;
  final String organizer;
  final OpportunityDetail opportunityDetail;
  final String createdAt;
  final String updatedAt;

  Opportunity.fromJson(Map<String, dynamic> jsonMap):
      id = jsonMap['id'] ?? 0,
      title = jsonMap['title'] ?? 0,
      category = jsonMap['category']!=null
          ? Category.fromJson(jsonMap['category'])
          : null,
      country = jsonMap['country']!=null
          ? Country.fromJson(jsonMap['country'])
          : null,
      deadline = jsonMap['deadline'] ?? "",
      organizer = jsonMap['organizer'] ?? "",
      opportunityDetail = jsonMap['opportunityDetails']!=null
          ? OpportunityDetail.fromJson(jsonMap['opportunityDetails'])
          : null,
      createdAt = jsonMap['createdAt'] ?? "",
      updatedAt = jsonMap['updatedAt'] ?? "";

}